watertap/unit_models/tests/test_gac.py

```diff
@@ -367,7 +364,96 @@ def test_solution_robust(self, gac_frame_robust):
         assert pytest.approx(12830000, rel=1e-3) == value(m.fs.unit.elap_time)
         assert pytest.approx(10.68, rel=1e-3) == value(m.fs.unit.bed_area)
 
-    @pytest.mark.requires_idaes_solver
+    @pytest.mark.component
+    def test_reporting_robust(self, gac_frame_robust):
+        m = gac_frame_robust
+        m.fs.unit.report()
+
+    @pytest.mark.component
+    def test_variable_sensitivity_robust(self, gac_frame_robust):
+        m = gac_frame_robust
+
+        import idaes.core.util.scaling as iscale
+        from pyomo.common.collections import ComponentSet
+
+        def var_sens_generator(
+            blk,
+            lb_scale=1e-1,
+            ub_scale=1e3,
+            tol=1e2,
+            zero=1e-10,
+        ):
+            test_scale = [lb_scale, ub_scale]
+            var_hist = {}
+            for scale in test_scale:
+                temp_blk = blk.clone()
+                # loop through variables and scale previously fixed inlet flow
+                for vt, vm in temp_blk.component_data_iterindex(
```

Comment:
`component_data_objects` outputs `vm` in this function. However this variable object is not directly application with                         `set_default_scaling(attrbute, value, index=None)` because the components are indexed individually and at different values. For example, `component_data_iterindex` will output the following `('flow_mol_phase_comp', ('Liq', 'H2O')) fs.unit.process_flow.properties_out[0.0].flow_mol_phase_comp[Liq,H2O]`. If vm or the `component_data_objects` was solely available, you could not identify the correct variable in the default scaling function as `temp_blk.fs.properties.set_default_scaling("flow_mol_phase_comp", dsf * scale**-1, ('Liq', 'H2O'))`. I found this function to be easiest to gain the variable and index in the appropriate string form.

Reply:
@hunterbarber Could you clarify exactly what you are trying to do here? Without comments or documentation, I am struggling to understand what is going on (especially seeing this is part of a test).

Reply:
So where `badly_scaled_var_generator` outputs which variables are outside of a properly scaled tolerance for a solved frame, it allows for the model dev to set scaling factors in the `calculate_scaling_factors` function to be well posed for the data frames used in testing. Though, I found this mainly lends itself to setting constant scaling factors in initial model development with whatever process scale are used in the tests. What this method intended to do was to take a previously defined and solved model frame in testing that has fixed inlet flow and default scaling factors. Then, it would downscale and upscale the process inlet flowrate and default scaling factors to approximate small-scale and large-scale limitations of the process. By varying the process scale and reporting values and scaled values (with initial constant scaling factors) of all model variables, the tool can show variables that are linearly or nonlinearly proportional (or inversely proportional) to the inlet flow. These are then good candidates to have non-constant scaling factors or user defined scaling factors in `calculate_scaling_factors`. Specifically, I had some scaled variables that changed between 100 and 0.01 in different model frames that weren't caught by `badly_scaled_var_generator` but were caught by this tool by changing in orders of magnitude across different process scales.

Reply:
First, this sounds like it needs to be a separate function with a set of dedicated tests to ensure it does what is expected. This might also be something that should go in IDAES rather than WaterTap.

Secondly however, I would be weary about using this in a test (unless it is a test for the method) - tests are generally best written in concrete terms, where you have control over as much as possible.